spec/models/manageiq/providers/amazon/aws_refresher_spec_common.rb

```diff
@@ -880,18 +867,8 @@ def assert_specific_vm_on_cloud_network_public_ip
     expect(v.network_ports.first.fixed_ip_addresses).to match_array([@ip2.fixed_ip_address])
     expect(v.network_ports.first.ipaddresses).to match_array([@ip2.fixed_ip_address, @ip2.address])
     expect(v.ipaddresses).to match_array([@ip2.fixed_ip_address, @ip2.address])
-
-    if options.inventory_object_refresh
-      expect(v.operating_system).to(
-        have_attributes(
-          :product_name => "linux_redhat",
-        )
-      )
-    else
-      # Old refresh can't fetch the public image and there fore also operating_system
-      expect(v.operating_system).to be_nil
-    end
-
+    expect(v.operating_system).to(have_attributes(:product_name => "linux_redhat"))
+    expect(v.operating_system).to be_nil
```

Comment:
I don't think this can be both `{:product_name => "linux_redhat"}` _and_ `nil`...